### ISSUE TITLE
False positive in rule parameter_in_package

### ISSUE THREAD

**Issue by skjdbg** [dalance/svlint#309]
In this ecample:
```systemverilog
package foo;
   class bar #( parameter int baz );
   endclass
endpackage
```

The rule "parameter_in_package" fails, but from the LRM:
> Unlike nonlocal parameters, local parameters can be declared in a generate block, package, class body, or
compilation-unit scope. In these contexts, the parameter keyword shall be a synonym for the localparam
keyword.

So a `parameter` should only fail this rule if it is inside a class body, not if it is inside a `parameter_port_list` of a class.

If I understand well the rule's implementation, any `parameter` within the scope of a package will fail this rule:
```rust
match node {
    RefNode::PackageDeclaration(x) => {
        let param = unwrap_node!(*x, ParameterDeclaration);
        if let Some(param) = param {
            let param_locate = unwrap_locate!(param).unwrap();
            SyntaxRuleResult::FailLocate(*param_locate)
        } else {
            SyntaxRuleResult::Pass
        }
    }
    _ => SyntaxRuleResult::Pass,
}
```
